### ISSUE TITLE
lifecycle with aws_instance causes diffs didn't match during apply

### ISSUE THREAD

**Issue by davedash** [hashicorp/terraform#3739]
When I add the following lifecycle to an `instance`:

```
resource "aws_instance" "instance" {
  ami = "${var.ami}"  # Ubuntu HVM
  instance_type = "${var.instance_type}"

  tags {
    Name = "${var.env}-${var.name}${var.name_suffix}"
    env = "${var.env}"
    role = "${var.env}-${var.name}"
  }

  ebs_block_device {
    encrypted = 1
    device_name = "/dev/sdf"
    volume_type = "gp2"
    volume_size = 50
  }

  security_groups = ["${split(",", var.security_groups)}"]

  subnet_id = "${element(split(",", var.subnets), var.number)}"

  iam_instance_profile = "${var.env}-${var.name}"
  ebs_optimized = "${var.ebs_optimized}"
  user_data = "${file(var.user_data)}"
  key_name = "davedash"

  lifecycle {
    ignore_changes =["user_data"]
  }

}
```

I get the following after a terraform plan/apply:

```
Error applying plan:

1 error(s) occurred:

* aws_instance.instance: diffs didn't match during apply. This is a bug with Terraform and should be reported.

Terraform does not automatically rollback in the face of errors.
Instead, your Terraform state file has been partially updated with
any resources that successfully completed. Please address the error
above and apply again to incrementally change your infrastructure.
```

I'll branch my code so I can pull out any other relevant details.

**Comment by phinze** [comment]
This might be related to #2627

**Comment by spiffxp** [comment]
Running into this exact problem as well with terraform 0.6.8

**Comment by phinze** [comment]
Hi folks! Some causes of `diffs didn't match during apply` were fixed up in #4965 - I'm attempting to tackle the remainder of known issues in #5627, so head there if you are still seeing troubles with `ignore_changes` in v0.6.12.
